### PR TITLE
SwatchColorPicker: let height and width be customizable

### PR DESCRIPTION
**Pull request by phkuo**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Let height, width, borderWidth, and cellMargins be customizable for the SwatchColorPicker.
Also modified interaction states to match latest redlines, and cleaned up lots of styles.

Before:
image
image

After:
image
image
(this last circles image has a customized larger height/width)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/SwatchColorPicker/ColorPickerGridCell.styles.ts [hunk 1]
```diff
@@ -1,110 +1,64 @@
...
-    borderColor: borderColor,
-    boxShadow: isHover ? 'none' : '0 0 0 1px #969696',
-    padding: 4,
-    margin: 0
-  };
-}
-
 export const getStyles = (props: IColorPickerGridCellStyleProps): IColorPickerGridCellStyles => {
-  const { theme, disabled, selected, circle, isWhite } = props;
+  const { theme, disabled, selected, circle, isWhite, height = 20, width = 20, borderWidth = 2 } = props;
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/SwatchColorPicker/ColorPickerGridCell.styles.ts]: Shouldn't these default values be in `defaultProps` static member of the `ColorPickerGridCell` class? Especially that it already has some default props in it.

File: packages/office-ui-fabric-react/src/components/SwatchColorPicker/ColorPickerGridCell.types.ts [hunk 2]
```diff
@@ -149,6 +164,21 @@ export interface IColorPickerGridCellStyleProps {
    * Whether the color being rendered is white or not. If it is white we show a border around it.
    */
   isWhite?: boolean;
+
+  /**
+   * The height of this cell, in pixels.
+   */
+  height?: number;
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/SwatchColorPicker/ColorPickerGridCell.types.ts]: Add `@default` values for documentation please for all the new props which have a default value.

File: packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.types.ts [hunk 3]
```diff
@@ -91,6 +91,26 @@ export interface ISwatchColorPickerProps {
    */
   doNotContainWithinFocusZone?: boolean;
 
+  /**
+   * The distance between cells, in pixels
+   */
+  cellMargin?: number;
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.types.ts]: Same... add `@default` for all new props.

File: packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.styles.ts [hunk 4]
```diff
@@ -6,17 +6,23 @@ const GlobalClassNames = {
 };
 
 export const getStyles = (props: ISwatchColorPickerStyleProps): ISwatchColorPickerStyles => {
-  const { className, theme } = props;
+  const { className, theme, cellMargin = 10 } = props;
```
Comment (Vitalius1) [inline packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.styles.ts]: Same... move this to the `defaultProps` static member of the `SwatchColorPicker` class

**Comment by phkuo** [comment]
> Also  for a great style cleanup:
> screen shot 2018-09-26 at 6 18 00 pm

most of that is because i removed the duplicate example :rofl:

**Comment by phkuo** [comment]
@betrue-final-final for screener approval. New redlines are from Janet.

**Pull request merged**